packages/Popover/src/Popover.js

```diff
@@ -296,11 +308,64 @@ class Popover extends React.Component {
     }
   };
 
+  handleKeyDown = event => {
+    if (event.key === "Tab" && this.isOpen() && this.$trigger) {
+      const isFocusOnFirst = this.focusIsOnCertainElementInPopover("first") || document.activeElement === this.$content;
+      const isFocusOnLast = this.focusIsOnCertainElementInPopover("last");
+
+      if (event.shiftKey && isFocusOnFirst) {
+        event.preventDefault();
+        this.focusableElements[this.triggerFocusIndex].focus();
+      } else if (!event.shiftKey && isFocusOnLast) {
+        event.preventDefault();
+        this.focusableElements[this.triggerFocusIndex + 1].focus();
+      }
+    }
+  };
+
+  focusIsOnCertainElementInPopover = which => {
+    const focusableElementsInPopover = this.$content.querySelectorAll(focusableElementSelector);
+    const index = which === "first" ? 0 : focusableElementsInPopover.length - 1;
+
+    return document.activeElement === focusableElementsInPopover[index];
+  };
+
+  elementIsDescendentOfPopover = elem => {
+    let elementsParent = elem.parentNode;
+    while (elementsParent !== null) {
+      if (elementsParent.getAttribute && elementsParent.getAttribute("data-pka-anchor") === "popover.content") {
+        return true;
+      }
+      elementsParent = elementsParent.parentNode;
+    }
+    return false;
+  };
+
+  restoreTabIndexes = () => {
+    this.focusableElements.forEach(focusableElement => {
+      if (focusableElement !== this.$trigger && !this.elementIsDescendentOfPopover(focusableElement)) {
+        focusableElement.tabIndex = this.originalTabIndexes.shift(); // eslint-disable-line no-param-reassign
+      }
+    });
+  };
+
+  removeTabIndexes = () => {
+    this.focusableElements.forEach((focusableElement, index) => {
+      if (focusableElement === this.$trigger) {
+        this.triggerFocusIndex = index;
+      } else if (!this.elementIsDescendentOfPopover(focusableElement)) {
+        this.originalTabIndexes.push(focusableElement.tabIndex);
+        focusableElement.tabIndex = -1; // eslint-disable-line no-param-reassign
```

Comment:
Did you try out removing the `tabindex` manipulation? This is the part that makes me the most nervous (this individual component alters a lot of external DOM elements, which isn't very component-like).

Reply:
Latest push does not touch the other items' tabIndex